src/al.jl

```diff
@@ -79,14 +78,15 @@ function al(nlp :: AbstractNLPModel; max_iter :: Int = 1000, max_time :: Real =
 
     # solve subproblem
     S = with_logger(NullLogger()) do
-      tron(al_nlp, x = x)
+      tron(al_nlp, x = copy(al_nlp.x))
     end
-    x = S.solution
-    cx = cons(nlp, x)
-    normcx = norm(cx)
+
+    al_nlp.x = S.solution
+    cons!(nlp, al_nlp.x, al_nlp.cx)
```

Comment:
Could you check if this is necessary? I think the last `x` was used to compute something and thus both `al_nlp.x` and `al_nlp.cx` are correct.

Reply:
I checked this with some impressions. I think you are right, al_nlp.x and al_nlp.cx are correct.